src/olympia/users/tests/test_models.py

```diff
@@ -795,26 +795,83 @@ def test_blocked_user_login_ip(self):
         request = RequestFactory(REMOTE_ADDR='192.168.0.8').get('/')
         request.user = user_factory(last_login_ip='192.168.1.1')
         IPNetworkUserRestriction.objects.create(network='192.168.1.1/32')
-        assert not IPNetworkUserRestriction.allow_request(request)
+        assert not IPNetworkUserRestriction.allow_submission(request)
+
+    def test_allowed_approval_while_blocking_submission(self):
+        request = RequestFactory(REMOTE_ADDR='192.168.0.1').get('/')
+        request.user = user_factory(last_login_ip='10.0.0.1')
+        IPNetworkUserRestriction.objects.create(network='192.168.0.0/28')
+        # Submission is allowed.
+        assert not IPNetworkUserRestriction.allow_submission(request)
+        # Approval is blocked.
+        upload = FileUpload.objects.create(ip_address='192.168.0.1', user=request.user)
+        assert IPNetworkUserRestriction.allow_auto_approval(upload)
+
+    def test_blocked_approval_last_login_ip(self):
+        request = RequestFactory(REMOTE_ADDR='192.168.0.1').get('/')
+        request.user = user_factory(last_login_ip='192.168.0.1')
+        IPNetworkUserRestriction.objects.create(
+            network='192.168.0.0/28', restriction_type=RESTRICTION_TYPES.APPROVAL
```

Comment:
Does the restriction on `192.168.0.0` also encompass `192.168.0.1`? If not, I don't follow how adding the restriction on `192.168.0.0` actually blocks this at all? If it does, then the test data on line 820 seems invalid because `192.168.0.2` would be blocked anyway.

Reply:
`192.168.0.0/28` should affect `192.168.0.1` to `192.168.0.14`. You're right that `192.168.0.2` should be changed..